host-components/README.md

```diff
@@ -0,0 +1,126 @@
+wasmtime.
+ - The demo currently has little support for outputs from a workload.
+ - The demo currently doesn't support filesystem access from a workload.
+ - The demo currently doesn't support socket access from a workload.
+ - The demo currently has a hard-coded .wasm (WebAssembly) file to run.
+ Others can be used, but do not expect much output!
+
+JSON is used for communications.
+Communications to the keep-manager from the keep-manager-tester are over HTTPS.
+Communications to the app-loader from the app-loader-tester are over HTTPS
+with both client and server certificates (the latter being created and self-
+signed by the app-loader - see note above about non-localhost hosts).
+
+systemd is used to create Keeps that can survive longer than the lifetime of
+the keep-manager process.  Alternatives to systemd are currently not supported.
+
+In order to run the demo, you shouldn't need any tools beyond those normally
+required to compile Enarx, but you do require sudo or administrator access
+(to set up the systemd components).
+
+Steps:
+1. download the latest source code from github (the easiest way is to do a
+`git clone` of the repository.
+2. enter the enarx/keep-runtime directory
+3. run `cargo build`
+4. edit the file `enarx/host-components/src/lib.rs` to reflect the location
+on your system of the `keep-runtime` binary created in the previous
+step (it should be have been created in
+`$YOUR_ROOT/enarx/keep-runtime/target/x86_64-unknown-linux-musl/debug/keep-runtime`)
+5. enter the enarx/host-components directory
+6. run `cargo build`
+7. edit the file `enarx/host-components/external/enarx-keep@.service` to
+update the `ExecStart` entry to reflect the location of the `keep-loader`
+binary created in the previous step
+8. edit the file `enarx/host-components/external/enarx-keep@.service` to
+update the `StandardOutput` and `StandardError` entries to reflect your
+preferred locations for the files to record stdio and stderr
+9. run `sudo ln -s enarx-keep@.service /etc/systemd/user/enarx-keep@.service`
+10. run `sudo systemd reload-daemon` (running this with the parameter `--user`
+**may** work)
+
+If you have run the demo before, you should kill old instances
+of the keep-loader.  This is best down by running the command
+`pkill -9 keep-loader`.  You may also wish to delete any keeploader output
+files (the locations of which you set in step 8).  Although it's unlikely to
+have any impact on the demo, you may wish to clean up old files with
+`rm /tmp/enarx-keep*.sock`.
+
+**NOTE** if you make any changes to the keep-loader.rs file, after recompiling
+it, you will need to remove the old link in /etc/systemd/user/, recreate it
+(step 10 above), and perform reload the systemctl daemon (step 11 above).
+
+## Running the demo
+
+You will need two different command line terminals, both running on the same
+machine.
+
+Terminal 1 - server:
+ - enter the `enarx/host-components` directory
+ - run the command `.target/debug/keep-manager`
```

Comment:
I prefer:

```console
$ cargo run --bin keep-manager
```

which ensures, that you actually built the stuff :smile:

Reply:
Fair.  The reason I moved to the alternative was frankly because of the number of warnings I was getting each time I did that.  Those warnings are now fixed, so changing this!